packages/flutter/lib/src/rendering/debug.dart

```diff
@@ -283,7 +280,6 @@ bool debugAssertAllRenderVarsUnset(String reason, { bool debugCheckIntrinsicSize
         debugPrintMarkNeedsLayoutStacks ||
         debugPrintMarkNeedsPaintStacks ||
         debugPrintLayouts ||
-        debugCheckIntrinsicSizes != debugCheckIntrinsicSizesOverride ||
```

Comment:
I'm not really sure about this change here. Currently, `debugCheckIntrinsicSizesOverride` depends on the value set in the `checkIntrinsicSizes` instance variable of `TestWidgetsFlutterBinding`. Even if we make `TestWidgetsFlutterBinding.checkIntrinsicSizes` a setter, we will still need to initialize the binding before changing the value in `flutter_test_config.dart`. Some tests in the framework do not use `TestWidgetsFlutterBinding` though, so initializing `TestWidgetsFlutterBinding` and then changing the value of `TestWidgetsFlutterBinding.checkIntrinsicSizes` would change the semantics of the tests.

If we're directing users to set `debugCheckIntrinsicSizes` from `package:flutter/rendering.dart`, then this check here should probably be removed. Another option would be to introduce a new nullable top level setter in `package:flutter_test`, and change the logic of `TestWidgetsFlutterBinding.checkIntrinsicSizes` to 

```dart
bool get checkIntrinsicSizes => newTopLevelFlutterTestcheckIntrinsicSizes ?? false;
```

We could even deprecate the `TestWidgetsFlutterBinding.checkIntrinsicSizes` to make it such that there's only one way to turn on the checks (via `debugCheckIntrinsicSizes` from `package:flutter/rendering.dart`) and keeping it simpler. This might not even be a breaking change because it's not used in Google3 and probably not by customer tests (need to verify this).

Reply:
The purpose of these checks is to make sure that someone doesn't set the flag in a test, then forget to unset it, so that it affects all subsequent tests.

I don't mind exactly how we do that. For most other flags we just check that it's not set, but that would mean that we couldn't override it globally. We could store the value at the start of the test, then check here that it matches the value at the start of the test. That would still catch people who fiddle with the flag in a test, without preventing the use of the flag as a global override.

Reply:
> We could store the value at the start of the test, then check here that it matches the value at the start of the test

Updated `_verifyInvariants` in `TestWidgetsFlutterBinding` to do this.